### DISCUSSION TITLE
Accessing the field from a related collection

### DISCUSSION THREAD

**Discussion by jsx101** [directus/directus#15922] [Archive]
Hi, I'm new to all this and am trying to create an M2O interface extension that will also show selected fields and values from the related collection. 

When creating it, I would like to have a field like Display Template here that shows a dropdown of all the fields from the related collection for me to select and pass as an array to the Vue component props (I'm also new to Vue but I hope this part makes sense).
![image](https://user-images.githubusercontent.com/40978303/194753078-73e41896-86bf-4270-8e43-c5c1871217bd.png)

I've tried referencing the index.ts files for select-dropdown-m2o's Display Template field. I doubt the "interface" value here is supposed to be "system-display-template". I've tried replacing it with "list" but that doesn't seem to work.
![image](https://user-images.githubusercontent.com/40978303/194754229-ff497373-e6f9-422f-b2b8-f5d22f713891.png)

Side question: what are the string values I can use for the "interface" field? I've tried following the code to find the type but for this field, its type is just "string | null" in FieldMeta type.

**Comment by azrikahar** [comment]
Can you share your custom interface code? It seems like you shared a piece of code from here:

https://github.com/directus/directus/blob/4904c3bba1e360476e316a7d2fc06a5dc10a5409/app/src/interfaces/select-dropdown-m2o/index.ts#L19-L28

which is for interface _options_ (for configuring the interface), but it sounds like you may want to use `<system-display-template>` directly in your custom interface's template. You should be able to use the `<v-field-template>` instead since system-display-template wraps v-field-template:

https://github.com/directus/directus/blob/79be789257c0aeb5e7c93b53eee93c90994c66c5/app/src/interfaces/_system/system-display-template/system-display-template.vue#L7-L13

> I've tried replacing it with "list" but that doesn't seem to work.

Note that the code you were referring to were for interface options, not the interface itself. Example of interface options _for_ WYSIWYG interface would be `Toolbar`, `Font`, `Folder` etc:

![chrome_vDnCVe4DxS](https://user-images.githubusercontent.com/42867097/195032426-a50ade49-2d03-4a15-a8fb-98eb43f2bb5f.png)

but the interface itself is just a TinyMCE editor in a _form_:

![image](https://user-images.githubusercontent.com/42867097/195032776-6550badf-5fab-4964-b11c-a0b995178cde.png)

so if you were expecting list to show up in a form, then it would be if you are only using it interface options.

> Side question: what are the string values I can use for the "interface" field? I've tried following the code to find the type but for this field, its type is just "string | null" in FieldMeta type.

you should be able to use any interface in [this directory](https://github.com/directus/directus/tree/main/app/src/interfaces), such as `list`, `boolean` etc, and any system interfaces in the `app/src/interfaces/_system` directory.

**Reply by jsx101** [reply]
Hi, thanks for your reply. Yes, it is the interface options that I am working on for now. 

I would like to have an interface option to choose a related collection, and when that collection is chosen, I would like to have another interface option for me to select a set of fields from that related collection.

My goal is to create an interface in the form with many input fields. One to select an item from the related collection, and a varying number of other input fields that are mapped one-to-one to fields that exist in the related collection. After selecting an item from the related collection, the other input fields will be auto-filled with values for that item.


> Can you share your custom interface code? It seems like you shared a piece of code from here

Yes, I am referencing code from the repo. The "Display Template" interface option is the only one I can find that shows a dropdown of the related collection's fields, so I copied and tried tweaking it to be able to get a list of fields I want to send to the interface component later on if that makes sense.
